gflownet/utils/buffer.py

```diff
@@ -200,33 +200,55 @@ def make_data_set(self, config):
         """
         if config is None:
             return None, None
-        elif "path" in config and config.path is not None:
-            path = self.logger.logdir / Path("data") / config.path
-            df = pd.read_csv(path, index_col=0)
-            # TODO: check if state2readable transformation is required.
-            return df
-        elif "type" not in config:
+        print("\nConstructing data set ", end="")
+        if "type" not in config:
             return None, None
+        elif config.type == "pkl" and "path" in config:
+            print(f"from pickled file: {config.path}\n")
+            with open(config.path, "rb") as f:
+                data_dict = pickle.load(f)
+                samples = data_dict["x"]
+                n_samples_orig = len(samples)
+                print(f"The data set containts {n_samples_orig} samples", end="")
+                samples = self.env.process_data_set(samples)
+                n_samples_new = len(samples)
+                if n_samples_new != n_samples_orig:
+                    print(
+                        f", but only {n_samples_new} are valid according to the "
+                        "environment settings. Invalid samples have been discarded."
+                    )
+                samples = samples[:20]
+                print("We are currently selecting only 20 samples")
+                print("Remember to write a function to normalise the data in code")
+                print("Max number of elements in data set has to match config")
+                print("Actually, write a function that contrasts the stats")
```

Comment:
Next level reminder writing.

Reply:
@vict0rsch suggestion haha I must say it's a good one haha